src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -91,6 +124,38 @@ export const Configurator = ({
     fetchFields();
   }, [baseDataset, indexPatternsService, queryString, timeFields.length]);
 
+  const updateDatasetForIndexedView = useCallback(async () => {
+    if (!indexedViewsService || !selectedIndexedView) {
+      return dataset;
+    }
+
+    let connectedDataSource;
+    if (dataset.dataSource?.id) {
+      const connectedDataSourceSavedObj: any = await indexedViewsService.getConnectedDataSource(
+        dataset
+      );
+      if (connectedDataSourceSavedObj) {
+        connectedDataSource = {
+          id: connectedDataSourceSavedObj.id,
+          title: connectedDataSourceSavedObj.attributes?.title,
+          type: 'DATA_SOURCE',
+        };
+      }
+    }
+
+    return {
+      ...dataset,
+      id: `${dataset.id}.${selectedIndexedView}`,
+      title: selectedIndexedView,
+      type: DEFAULT_DATA.SET_TYPES.INDEX,
+      sourceDatasetRef: {
+        id: dataset.id,
+        type: dataset.type,
+      },
+      dataSource: connectedDataSource ?? dataset.dataSource,
```

Comment:
apologies. still not able to understand the difference here. dataset.dataSource should always be the same as connectedDataSource. If this is required because title is not present that I think we should include the title within the `sourceDatasetRef` at the time of creation and all times rather than creating a new function

Reply:
The connected data source is the collection attached to the data-connection ds